### PR TITLE
snippet injection add smoke test

### PR DESCRIPTION
**Pull request by siyuniu-ms**
use the snippet-injection version otl and add dependency at application insights

### CODE REVIEW

**Comment by siyuniu-ms** [comment]
Things that could be improved:
1. Enable user have two connection string, one for server, and the other for browser. This could help when authentication is required under some cases.
2. Check the connection string to be set before setting the snippet, and provide alternative plan if sth went wrong.

**Comment by siyuniu-ms** [comment]
the github build failed because it could not get my local version of otl java.

**Comment by siyuniu-ms** [comment]
add smoke test

File: settings.gradle.kts [hunk 1]
```diff
@@ -125,6 +125,8 @@ hideFromDependabot(":smoke-tests:apps:RoleNameOverrides")
 hideFromDependabot(":smoke-tests:apps:RuntimeAttach")
 hideFromDependabot(":smoke-tests:apps:RuntimeAttachWithDelayedConnectionString")
 hideFromDependabot(":smoke-tests:apps:Sampling")
+hideFromDependabot(":smoke-tests:apps:SnippetInjection")
+hideFromDependabot(":smoke-tests:apps:SnippetInjection:MockExtension")
```
Comment (trask) [inline settings.gradle.kts:129]:
```suggestion
```

File: dependencyManagement/build.gradle.kts [hunk 2]
```diff
@@ -70,6 +70,7 @@ val CORE_DEPENDENCIES = listOf(
   "io.opentelemetry.javaagent:opentelemetry-javaagent-extension-api:${otelInstrumentationAlphaVersion}",
   "io.opentelemetry.javaagent:opentelemetry-javaagent-bootstrap:${otelInstrumentationAlphaVersion}",
   "io.opentelemetry.javaagent:opentelemetry-javaagent-tooling:${otelInstrumentationAlphaVersion}",
+  "io.opentelemetry.javaagent.instrumentation:opentelemetry-javaagent-servlet-common-bootstrap:${otelInstrumentationAlphaVersion}",
```
Comment (trask) [inline dependencyManagement/build.gradle.kts:73]: is this needed in this PR?

**Review by trask — APPROVED** [review]

**Pull request merged**